openedx/adg/admin_dashboard/webinars/forms.py

```diff
@@ -0,0 +1,138 @@
+"""
+Forms for webinars app.
+"""
+import datetime
+import json
+
+import pytz
+from django import forms
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from openedx.adg.common.lib.zoom_client.client import ZoomClient
+from openedx.adg.lms.webinars.constants import WEBINAR_DEFAULT_TIME_ZONE
+from openedx.adg.lms.webinars.models import Webinar
+
+from .constants import WEBINAR_OPTIONS
+
+
+class WebinarForm(forms.ModelForm):
+    """
+    Webinar Form to create/edit a webinar from admin dashboard
+    """
+
+    class Meta:
+        model = Webinar
+        fields = ['title', 'description', 'language', 'meeting_link', 'webinar_options',
+                  'start_time', 'duration', 'webinar_timezone',
+                  'audio_settings', 'enable_host_video', 'enable_panelist_video', 'banner',
+                  'invites_by_email_address', 'panelists']
+
+    def __init__(self, *args, **kwargs):
+        """
+        Override to initialize and clean the fields
+        """
+        super(WebinarForm, self).__init__(*args, **kwargs)
+        self.fields['meeting_link'].disabled = True
+
+        self.initialize_fields()
+        self.context = self.get_context_data()
+
+        if self.data:
+            self.data = self.data.copy()
+            self.data['webinar_options'] = self.format_webinar_options(self.data.getlist('webinar_options'))
+            self.data['start_time'] = self.format_start_time()
+            self.data['duration'] = datetime.timedelta(
+                hours=int(self.data.get('duration_hours', 0)), minutes=int(self.data.get('duration_mins', 0))
+            )
+            self.data.setlist(
+                'panelists', list(User.objects.filter(email__in=self.data.get('panelists', '').split(',')))
+            )
+
+    def format_webinar_options(self, user_selected_webinar_option):
+        """
+        Convert user selected webinar options into JSON format
+        """
+        webinar_options = {}
+        for option in WEBINAR_OPTIONS:
+            webinar_options[option] = option and (option in user_selected_webinar_option)
+        return json.dumps(webinar_options)
+
+    def format_start_time(self):
+        """
+        Convert user selected day/month/year/hour/min to a webinar start_time
+        """
+        day = self.data.get('start_time_day')
+        month = self.data.get('start_time_month')
+        year = self.data.get('start_time_year')
+        hour = self.data.get('start_time_hour')
+        minute = self.data.get('start_time_min')
+        timezone = self.data.get('webinar_timezone')
+        try:
+            date_time = datetime.datetime(int(year), int(month), int(day), int(hour), int(minute))
+            return pytz.timezone(timezone).localize(date_time)
+        except (ValueError, TypeError):
+            return None
```

Comment:
Any reason we are not using clean methods for this?

Reply:
As the form doesn't have start_time value so not adding start_time value before clean gives validation error.

Reply:
We'll be covering this as discussed in the call.